### ISSUE TITLE
Feat: Add possibility to use a different integrity field hashing algorithm (e.g. sha1)

### ISSUE THREAD

**Issue by PrettyCoffee** [lirantal/lockfile-lint#206]
**Is your feature request related to a problem? Please describe.**

At my workplace we have our own npm registry provided with ADOS (Azure DevOps Server). However, ADOS does only support sha1 integrity fields and not sha512 and it doesn't seem like they will update that.
This leaves us stuck at lockfile-lint version 4.8.0 right now.

**Describe the solution you'd like**

I think being able to provide an optional algorithm to `--validate-integrity` (which could default to sha512) would be nice, e.g.:

```sh
lockfile-lint --path ./package-lock.json --type npm --validate-integrity sha1 --validate-https --allowed-hosts <ados-server-domain>
```

**Describe alternatives you've considered**

1. Staying on 4.8.0
2. Updating to the latest version and dropping `--validate-integrity`

**Comment by lirantal** [comment]
Hmmm, right now that config flag doesn't receive a value, it's just a boolean so changing it wouldn't be as minimal as adding another option. Perhaps dropping integrity validation for now on your end would be the easiest path forward since it seems you anyway can't stay on a secure signature hash.

**Comment by PrettyCoffee** [comment]
I see, then we will probably drop the flag for now. Thanks a lot for the quick reply!
